hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java

```diff
@@ -131,7 +133,9 @@ public void run(Runnable task) {
     }
 
     private boolean publishCurrentTask() {
-        return (getPartitionId() != AD_HOC_PARTITION_ID && currentTask == null);
+        boolean isClientRunnable = currentTask instanceof MessageTask
```

Comment:
or does anybody know why `currentTask == null` is added/required? maybe we should just get rid of that check...